From 3c46a530f38d69818cd2a00b99c395f1a9113535 Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Fri, 10 Jun 2016 11:40:41 +0700
Subject: [PATCH] DoorPhone-iWave-Fix audio issue on outdoor script

At current time, iWave board can not output audio because the audio pipeline
use format S32LE for audio and iWave can not output audio with this format

This patch file modifies outdoor script to fix this issue
This patch file fix this issue by using format S16LE that iWave board can
use to output audio

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 OutDoor/run_outdoor_opencv.sh | 2 +-
 doorphone.init                | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/OutDoor/run_outdoor_opencv.sh b/OutDoor/run_outdoor_opencv.sh
index 9fd20dd..77c7c6a 100755
--- a/OutDoor/run_outdoor_opencv.sh
+++ b/OutDoor/run_outdoor_opencv.sh
@@ -3,7 +3,7 @@
 VIDEOLOCATION=/home/root/videos/
 SCRIPTLOCATION=/home/root/doorphone/    # Only effect in this file, remember to change other files as well
 gst-launch-1.0 alsasrc device=hw:0,0 ! audioconvert ! avenc_aac compliance=-2 ! rtpmp4apay ! udpsink host=192.168.10.100 port=2000 sync=false &> /dev/null &
-gst-launch-1.0 udpsrc port=2001 caps="application/x-rtp,config=(string)40002420" ! rtpjitterbuffer ! queue ! rtpmp4adepay ! avdec_aac ! audioconvert !  alsasink &> /dev/null &
+gst-launch-1.0 udpsrc port=2001 caps="application/x-rtp,config=(string)40002420" ! rtpjitterbuffer ! queue ! rtpmp4adepay ! avdec_aac ! audioconvert ! audio/x-raw,format=S16LE ! alsasink &> /dev/null &
 
 # Send video stream to Base phone and Out door
 gst-launch-1.0 -e v4l2src ! video/x-raw,format=UYVY,width=720,height=480 ! tee name=t t. ! vspmfilter outbuf-alloc=true ! video/x-raw,format=NV12,width=640,height=480 ! omxh264enc target-bitrate=10485760 num-p-frames=0 ! h264parse ! video/x-h264,stream-format=avc,alignment=au ! rtph264pay pt=96 config-interval=3 mtu=9000 ! udpsink host=192.168.10.255 port=1234 t. ! vspmfilter outbuf-alloc=true ! video/x-raw,format=NV12,width=736,height=480 ! omxh264enc target-bitrate=10485760 num-p-frames=0 ! h264parse ! video/x-h264,stream-format=avc,alignment=au ! rtph264pay pt=96 config-interval=3 ! udpsink host=192.168.10.101 port=5555 &> /dev/null &
diff --git a/doorphone.init b/doorphone.init
index 8472d96..2b5e323 100755
--- a/doorphone.init
+++ b/doorphone.init
@@ -4,7 +4,7 @@ case "$1" in
 	start)
 		export LD_LIBRARY_PATH="/lib:/usr/lib:/usr/local/lib:"
 		amixer set "LINEOUT Mixer DACL" on
-		amixer set "DVC Out" 15
+		amixer set "DVC Out" 100
 		amixer set "DVC In" 15
 		amixer set "Digital" 200,200
 		modprobe -a mmngr mmngrbuf s3ctl uvcs_cmn vspm fdpm vsp2
-- 
1.9.1

